Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/JobDataInteraction/DataStreamActiveReader.cs

```diff
@@ -25,9 +26,10 @@ internal DataStreamActiveReader(NativeArray<EntityKeyedTaskWrapper<TInstance>> a
         /// Gets the <typeparamref name="TInstance"/> at the specified index.
         /// </summary>
         /// <param name="index">The index into the backing array</param>
-        public TInstance this[int index]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ref readonly TInstance ElementAtReadOnly(int index)
         {
-            get => m_Active[index].Payload;
+            return ref m_Active.ElementAtReadOnly(index).Payload;
         }
```

Comment:
Should we be dropping support for the indexer? I know most of the time getting the `ref` is more efficient but there are cases where it's not.
Also, if you want to call a bunch of methods non readonly methods on a mutable type then you'll want to convert this out of a `ref` anyway to prevent defensive copies at each call.

plus, we expose the indexer getter for the writers/updaters. Seems like we should keep it here for the sake of consistency.